model/schema/controlledVocabulary.yaml

```diff
@@ -80,3 +73,11 @@ classes:
       name:
         required: true
         identifier: true
+
+  VocabularyTermSynonym:
+    is_a: Synonym
+    description: >-
+      The relationship between a controlled vocabulary term and a synonym.
+    slot_usage:
+      subject:
+        range: VocabularyTerm
```

Comment:
The VocabularyTerm is the domain, not the range - same comment goes for OntologySynonym and ResourceSynonym.  Also, I don't see any slots linking these classes to the VocabularyTerm / Ontology / Resource.

Do you envisage that there will eventually be different slots for the synonyms of each of these classes, or could we just use a multivalued 'synonyms' slot that can be added to all these classes and has a range of Synonym?

Reply:
Maybe I'm confused, but I've looked at this again, and as written, I think the subject in these cases is correct. For example, we have a VocabularyTermSynonym object (not a slot of a VocabularyTerm, but its own object), and the subject slot for that object (to which the synonym applies) must refer to a VocabularyTerm.

Also, because these Synonym objects (AlleleSynonym, ResourceSynonym, etc) already have a subject slot that points to the relevant entity (e.g., gene, allele, resource, etc), I had not envisioned a need for a synonyms slots in the entities that have synonyms. 

I'm guessing that my proposal is somewhat different from what you had in mind? I chose this path because there are many-to-many entity-synonym relationships, and it seemed better to specify the entity (e.g., vocabulary term) to which the synonym applies right here in the Synonym entity.